pkg/instrumentation/dotnet.go

```diff
@@ -33,11 +33,14 @@ const (
 	envDotNetOTelAutoHome               = "OTEL_DOTNET_AUTO_HOME"
 	dotNetCoreClrEnableProfilingEnabled = "1"
 	dotNetCoreClrProfilerID             = "{918728DD-259F-4A6A-AC2B-B85E1B658318}"
-	dotNetCoreClrProfilerPath           = "/otel-auto-instrumentation/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so"
+	dotNetCoreClrProfilerGlibcPath      = "/otel-auto-instrumentation/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so"
+	dotNetCoreClrProfilerMuslPath       = "/otel-auto-instrumentation/linux-musl-x64/OpenTelemetry.AutoInstrumentation.Native.so"
 	dotNetAdditionalDepsPath            = "/otel-auto-instrumentation/AdditionalDeps"
 	dotNetOTelAutoHomePath              = "/otel-auto-instrumentation"
 	dotNetSharedStorePath               = "/otel-auto-instrumentation/store"
 	dotNetStartupHookPath               = "/otel-auto-instrumentation/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll"
+	dotNetGlibcLinuxRuntime             = "linux-glibc"
+	dotNetMuslLinuxRuntime              = "linux-musl"
```

Comment:
I propose to rename:
-  `dotNetGlibcLinuxRuntime` to `dotNetRuntimeLinuxGlibc`
-  `dotNetMuslLinuxRuntime` to `dotNetRuntimeLinuxMusl`

so that they have the same prefix which is the usual pattern for defining enumerations in Go.

Also the constants could be defined in a seperate `const ( )` block with a comment above the block saying that these are supported values for dotnet-runtime annoation. 

Example: https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/net/http/status.go;l=7-9